src/components/topic/TopicService.js

```diff
@@ -71,10 +71,15 @@ goog.require('ga_permalink');
         });
 
         // Returns a promise that is resolved when topics are loaded
-        this.getTopics = function() {
+        this.loadConfig = function() {
           return topicsP;
```

Comment:
Lines 65-71 should not be done with the topicsP promise. It should be done in the loadConfig functions on success, before resolving the topicsP promise. Then I think we avoid the race condition. (and it's more proper anyway).